### DISCUSSION TITLE
How do reconnect client

### DISCUSSION THREAD

**Discussion by MohMehrnia** [jchristn/WatsonTcp#301] [Q&A]
How do reconnect client when server restarted

**Comment by jchristn** [comment]
You can attach an event handler to `WatsonTcpClient.Events.ServerDisconnected` to be notified of the disconnection, and re-connect using `WatsonTcpClient.Connect`.

**Comment by ShayanFiroozi** [comment]
> How do reconnect client when server restarted

**Auto Reconnect** feature was developed and works fine , u can easily use it 🤘👍

**Reply by MohMehrnia** [reply]
how do used?

**Reply by ShayanFiroozi** [reply]
There's a pull request which has not been accepted by the repository maintainer yet , but you can use the fork on my repositories with Auto Reconnect feature.

**Reply by jchristn** [reply]
Hi @ShayanFiroozi I merged the PR, tested, and found an issue.  I reverted back, but left the merge in a new branch `v4.0-autorc` for you to examine.  Please refer to the PR comments for the issue I found.
